src/Functions/array/arrayIndex.h

```diff
@@ -635,39 +614,33 @@ class FunctionArrayIndex : public IFunction
     template <typename... Integral>
     static ColumnPtr executeIntegral(const ColumnsWithTypeAndName & arguments)
     {
-        const ColumnArray * const left = checkAndGetColumn<ColumnArray>(arguments[0].column.get());
-
-        if (!left)
+        const auto * array = checkAndGetColumn<ColumnArray>(arguments[0].column.get());
+        if (!array)
             return nullptr;
 
-        const ColumnPtr right_converted_ptr = arguments[1].column->convertToFullColumnIfLowCardinality();
```

Comment:
Another issue, that we could convert column to full several times: in `executeIntegral`, `executeString`...